compiler/rustc_traits/src/implied_outlives_bounds.rs

```diff
@@ -94,9 +94,11 @@ fn compute_implied_outlives_bounds<'tcx>(
         // region relationships.
         implied_bounds.extend(obligations.into_iter().flat_map(|obligation| {
             assert!(!obligation.has_escaping_bound_vars());
-            match obligation.predicate.kind() {
-                &ty::PredicateKind::ForAll(..) => vec![],
-                &ty::PredicateKind::Atom(atom) => match atom {
+            let binder = obligation.predicate.bound_atom();
+            if binder.skip_binder().has_escaping_bound_vars() {
+                vec![]
+            } else {
+                match binder.skip_binder() {
```

Comment:
this pattern seems unfortunate, it'd be nicer to match on `binder.no_bound_vars`

Reply:
Updated